PagingWithNetworkSample/app/src/main/java/com/android/example/paging/pagingwithnetwork/reddit/repository/inMemory/byPage/InMemoryByPageKeyRepository.kt

```diff
@@ -16,47 +16,35 @@
 
 package com.android.example.paging.pagingwithnetwork.reddit.repository.inMemory.byPage
 
-import androidx.annotation.MainThread
-import androidx.lifecycle.switchMap
-import androidx.paging.toLiveData
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.PagingDataFlow
 import com.android.example.paging.pagingwithnetwork.reddit.api.RedditApi
-import com.android.example.paging.pagingwithnetwork.reddit.repository.Listing
 import com.android.example.paging.pagingwithnetwork.reddit.repository.RedditPostRepository
 import com.android.example.paging.pagingwithnetwork.reddit.vo.RedditPost
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.flow.Flow
 import java.util.concurrent.Executor
 
 /**
  * Repository implementation that returns a Listing that loads data directly from network by using
  * the previous / next page keys returned in the query.
  */
-class InMemoryByPageKeyRepository(private val redditApi: RedditApi,
-                                  private val networkExecutor: Executor) : RedditPostRepository {
-    @MainThread
-    override fun postsOfSubreddit(subReddit: String, pageSize: Int): Listing<RedditPost> {
-        val sourceFactory = SubRedditDataSourceFactory(redditApi, subReddit, networkExecutor)
+class InMemoryByPageKeyRepository(
+        private val redditApi: RedditApi,
+        private val networkExecutor: Executor
+) : RedditPostRepository {
 
-        // We use toLiveData Kotlin extension function here, you could also use LivePagedListBuilder
-        val livePagedList = sourceFactory.toLiveData(
-                pageSize = pageSize,
-                // provide custom executor for network requests, otherwise it will default to
-                // Arch Components' IO pool which is also used for disk access
-                fetchExecutor = networkExecutor)
+    override fun postsOfSubreddit(subReddit: String, pageSize: Int): Flow<PagingData<RedditPost>> {
+        val sourceFactory = SubRedditPagingSourceFactory(
```

Comment:
you shouldn't need the factory class anymore

Reply:
Done